### PR TITLE
LPS-183464 The object field can be localized only when the object definition is custom and has enableLocalization true

### PR DESCRIPTION
**Pull request by seltong**
Related issue: [LPS-183464](https://issues.liferay.com/browse/LPS-183464)

### CODE REVIEW

**Review by MarinhoFeliphe — APPROVED** [review]
LGTM!! :tada:

**Comment by seltong** [comment]
@MarinhoFeliphe can I forward this pr?

**Comment by MarinhoFeliphe** [comment]
@seltong yes!

File: modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java [hunk 1]
```diff
@@ -300,12 +301,29 @@ protected ObjectDefinition randomObjectDefinition() throws Exception {
 						indexed = false;
 						indexedAsKeyword = false;
 						label = Collections.singletonMap("en_US", "Column");
+						localized =
+							objectDefinition.getEnableLocalization() &&
+							!objectDefinition.getSystem() &&
+							RandomTestUtil.randomBoolean();
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java:307]: @seltong I changed my mind and we don't need to put this `RandomTestUtil.randomBoolean()`

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

**Comment by MarinhoFeliphe** [comment]
Hey @seltong, see my comments

File: modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java [hunk 2]
```diff
@@ -300,12 +301,29 @@ protected ObjectDefinition randomObjectDefinition() throws Exception {
...
+					code = WorkflowConstants.STATUS_DRAFT;
+					label = WorkflowConstants.getStatusLabel(
+						WorkflowConstants.STATUS_DRAFT);
+					label_i18n = _language.get(
+						LanguageResources.getResourceBundle(
+							LocaleUtil.getDefault()),
+						WorkflowConstants.getStatusLabel(
+							WorkflowConstants.STATUS_DRAFT));
+				}
+			});
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java:326]: @seltong why the status needs to be DRAFT?

Comment (seltong) [reply modules/apps/object/object-admin-rest-test/src/testIntegration/java/com/liferay/object/admin/rest/resource/v1_0/test/ObjectDefinitionResourceTest.java:326]: Now we are looking at status.

```
@Override
protected String[] getAdditionalAssertFieldNames() {
	return new String[] {"name", "status"};
}
```

Therefore, we need to ensure that each created objectDefinition has the correct status and the default status is Draft.

**Comment by seltong** [comment]
ci:forward:force

**Comment by seltong** [comment]
ci:test:sf

**Comment by seltong** [comment]
ci:test:relevant

**Review by MarinhoFeliphe — APPROVED** [review]
LGTM!! :tada:

**Comment by MarinhoFeliphe** [comment]
ci:forward:force

**Pull request closed without merging**